src/storage/mvcc/txn.rs

```diff
@@ -282,104 +305,219 @@ mod tests {
         must_get(engine.as_ref(), b"x", 12, b"x5");
 
         must_gc(engine.as_ref(), b"x", 22);
+        must_get(engine.as_ref(), b"x", 22, b"x10");
         must_get_none(engine.as_ref(), b"x", 12);
 
         must_gc(engine.as_ref(), b"x", 32);
         must_get_none(engine.as_ref(), b"x", 22);
         must_get_none(engine.as_ref(), b"x", 40);
     }
 
-    fn to_fake_ts(ts: u64) -> u64 {
-        TEST_TS_BASE + ts
+    #[test]
+    fn test_write() {
+        let engine = engine::new_engine(Dsn::RocksDBPath(TEMP_DIR), DEFAULT_CFS).unwrap();
+
+        must_prewrite_put(engine.as_ref(), b"x", b"x5", b"x", 5);
+        must_seek_write_none(engine.as_ref(), b"x", 5);
+
+        must_commit(engine.as_ref(), b"x", 5, 10);
+        must_seek_write(engine.as_ref(),
+                        b"x",
+                        u64::max_value(),
+                        5,
+                        10,
+                        WriteType::Put);
+        must_reverse_seek_write(engine.as_ref(), b"x", 5, 5, 10, WriteType::Put);
+        must_seek_write_none(engine.as_ref(), b"a", u64::max_value());
+        must_reverse_seek_write_none(engine.as_ref(), b"y", 5);
+        must_get_commit_ts(engine.as_ref(), b"x", 5, 10);
+
+        must_prewrite_delete(engine.as_ref(), b"x", b"x", 15);
+        must_rollback(engine.as_ref(), b"x", 15);
+        must_seek_write(engine.as_ref(),
+                        b"x",
+                        u64::max_value(),
+                        15,
+                        15,
+                        WriteType::Rollback);
+        must_reverse_seek_write_none(engine.as_ref(), b"x", 15);
```

Comment:
Keys on CF_WRITE are in descending order, so reverse_seek_write will only return keys with greater ts.